tensorflow/python/keras/layers/convolutional.py

```diff
@@ -579,24 +600,41 @@ def __init__(self,
                kernel_constraint=None,
                bias_constraint=None,
                **kwargs):
+    self.checkProperKernel(kernel_size)
+    #if(rv==0):
+    
     super(Conv2D, self).__init__(
-        rank=2,
-        filters=filters,
-        kernel_size=kernel_size,
-        strides=strides,
-        padding=padding,
-        data_format=data_format,
-        dilation_rate=dilation_rate,
-        activation=activations.get(activation),
-        use_bias=use_bias,
-        kernel_initializer=initializers.get(kernel_initializer),
-        bias_initializer=initializers.get(bias_initializer),
-        kernel_regularizer=regularizers.get(kernel_regularizer),
-        bias_regularizer=regularizers.get(bias_regularizer),
-        activity_regularizer=regularizers.get(activity_regularizer),
-        kernel_constraint=constraints.get(kernel_constraint),
-        bias_constraint=constraints.get(bias_constraint),
-        **kwargs)
+          rank=2,
+          filters=filters,
+          kernel_size=kernel_size,
+          strides=strides,
+          padding=padding,
+          data_format=data_format,
+          dilation_rate=dilation_rate,
+          activation=activations.get(activation),
+          use_bias=use_bias,
+          kernel_initializer=initializers.get(kernel_initializer),
+          bias_initializer=initializers.get(bias_initializer),
+          kernel_regularizer=regularizers.get(kernel_regularizer),
+          bias_regularizer=regularizers.get(bias_regularizer),
+          activity_regularizer=regularizers.get(activity_regularizer),
+          kernel_constraint=constraints.get(kernel_constraint),
+          bias_constraint=constraints.get(bias_constraint),
+          **kwargs)
+    
+    
+    
+        
+  def checkProperKernel(self,kernel_size):
```

Comment:
why not use that in some base layer, or a regular function?

Reply:
I had implemented the function in base class Conv.